pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -118,3 +129,147 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command string, cluster cluste
+	for i,k := range containerConfig.HostConfig.VolumesFrom {
+		containerConfig.HostConfig.VolumesFrom[i] = m[k]			
+	}
+	// update affinity
+	if len(affinityRef) > 0 {
+		containerConfig.Env[affinityIndex] = "affinity:container==" + m[affinityRef[0]]
+	}
+	// update links
+	links = make([]string,0)
+	linkSet := make(map[string]bool) 
+	for _, link := range containerConfig.HostConfig.Links {
+		linkSplit := strings.SplitN(link,":",2)
+		containerIdName := strings.TrimSpace(linkSplit[0])
+		containerId := m[containerIdName]
+		if containerId != containerIdName {
+			linkSet[containerId + ":" + containerIdName] = true
+		}
+		if len(linkSplit) > 1 {
+			linkSet[containerId + ":" + strings.TrimSpace(linkSplit[1])] = true
+		}
+		for k,_ := range linkSet {
+			links = append(links,k)
+		}		
+	}
+	containerConfig.HostConfig.Links = links
+	return nil
+
+}
+func getIDsFromContainerReferences(cluster cluster.Cluster, tenantId string, containerReferences[] string, affinityLabelFirst bool ) (map[string]string, error) {
+	// containerReferences is a array of container references of the form long id, short id, or name
+	// if affinityLabelFirst is true the first element of containerReferences an affinity label reference of the form label:value
+	m := make(map[string]string)
+	for _, containerReference := range containerReferences {
+		m[containerReference] = ""
+	} 
+	for _, container := range cluster.Containers() {
+		if container.Labels[headers.TenancyLabel] == tenantId {
+		  for i, containerReference := range containerReferences {
+			var containerId string
+			if affinityLabelFirst && i == 0 {
+				containerId = getIDFromContainerLabel(container,tenantId,containerReference)
+			} else {
+				containerId = getContainerId(container,tenantId,containerReference)
+			}
+			if containerId != "" {
+				m[containerReference] = containerId
+			}			
+		  }
+		}		
+	}
+	for _, containerReference := range containerReferences {
+		if m[containerReference] == "" {
+			err := fmt.Errorf(NOTAUTHORIZED_ERROR, containerReference)
+			return nil,err
+		}
+	}
+	return m,nil
+}
+
+func getIDFromContainerLabel(container *cluster.Container, tenantId string, affinityLabelValue string ) (string) {
+	if container.Labels[headers.TenancyLabel] != tenantId {
+		return ""
+	}
+	// affinityLabelValue is in the form label==value
+	kv := strings.Split(affinityLabelValue,"==")
+	for k,v := range container.Config.Labels {
+		if k == kv[0] && v == kv[1] {
+			return container.Info.ID
+		} 
+	}
+	return ""	
+}
+
+func getContainerId (container *cluster.Container, tenantId string, containerReference string) (string) {
+	if container.Labels[headers.TenancyLabel] != tenantId {
+		return ""
+	}
+	// check for long id
+	if container.Info.ID == containerReference {
+		return container.Info.ID
+    } else {
+		// check for name
+	    for _, name := range container.Names {
+		   if (containerReference == name || containerReference == container.Labels[headers.OriginalNameLabel])  {
```

Comment:
Range always starts from the beginning of the array or map.  I'm not sure if there is a way to avoid this extra if.